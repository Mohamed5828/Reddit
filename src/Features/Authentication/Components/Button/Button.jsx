import React from "react";
import { BaseButton } from "./Button.styled";

/**
 * Button component that is used in authentication pages
 * @param {string} children The text of the button
 * @param {Array} otherProps Props to know the shape of the button 
 * @returns {React.Component}  Button component
 */
const Button = ({ children, ...otherProps }) => {
  return <BaseButton {...otherProps}>{children}</BaseButton>;
};

export default Button;
