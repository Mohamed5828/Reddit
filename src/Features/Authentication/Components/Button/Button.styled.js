import styled from "styled-components";

const fontFamily = "'IBM Plex Sans', sans-serif;";

export const BaseButton = styled.button`
  line-height: 40px;
  background-color: ${(p) =>
    p.page
      ? ({ theme }) => theme.color.link
      : ({ theme }) => theme.color.orange};

  border: none;
  cursor: pointer;
  display: flex;
  justify-content: center;
  font-family: Noto Sans, sans-serif;
  font-size: 14px;
  font-weight: 700;
  letter-spacing: unset;
  border-radius: ${(p) => (p.page ? "5px" : "999px")};
  color: ${({ theme }) => theme.color.baseButtonColor};
  height: 40px;
  padding: 0 16px;
  font-family: ${fontFamily};
  text-transform: unset;
  margin-top: 8px;
  width: 100%;

  opacity: ${(p) => (p.valid || p.page ? "1" : "0.3")};

  cursor: ${(p) => (p.valid || p.page ? "pointer" : "default")};

  &:hover {
    background-color: ${(p) =>
      !p.page
        ? p.valid
          ? ({ theme }) => theme.color.logInButtonColorHovered
          : ({ theme }) => theme.color.orange
        : ({ theme }) => theme.color.logInBlueHovered};
  }
`;
