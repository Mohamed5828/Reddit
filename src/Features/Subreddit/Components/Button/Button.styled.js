import styled from "styled-components";

export const StyledButton = styled.div`
  display: flex;


  .width {
    width: 96px;
  }

  button {
    width: 100%;
    position: relative;
    background-color: ${({joined, theme})=>joined?theme.background.primary:theme.color.secondary}   ;
    border: ${({joined, theme})=>joined?`1px solid ${theme.color.secondary}`:"none"};
    color: ${({joined, theme})=>joined?theme.color.secondary:theme.background.primary};
    font-family: "Noto Sans", Arial, sans-serif;
    font-size: 14px;
    font-weight: 700;
    letter-spacing: unset;
    line-height: 17px;
    text-transform: unset;
    min-height: 32px;
    min-width: 32px;
    padding: 4px 16px;
    border-radius:100px;
  }

  button:hover{
    background-color:${({joined})=>joined?"#f4f9fd":"#1483d6"} ;
  }

`;
