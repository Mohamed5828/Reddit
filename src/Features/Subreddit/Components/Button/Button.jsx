import { useState } from "react";
import { StyledButton } from "./Button.styled";

/**
 * 
 * @param {boolean} isJoined - boolean if joined community or not
 * @param {function} onJoin - function receive state of joining
 * @returns {React.Component} Join button
 */
const Button = ({isJoined, onJoin}) => {
  const [text, setText] = useState(isJoined ? "joined" : "join");

  /**
   * function handle click join button
   */
  const onClickJoin = () => {
    onJoin(text === "join" ? true : false);
    setText((prev) => (prev === "join" ? "joined" : "join"));
  };

  /**
   * function handle text when hover button
   */
  const onHover = () => {
    if (isJoined) {
      setText("leave");
    }
  };

  /**
   * function handle text when mouse out the button
   */
  const onLeave = () => {
    if (isJoined) {
      setText("joined");
    }
  };

  return (
    <StyledButton joined={isJoined}>
      <div className="width">
        <button onClick={onClickJoin} onMouseOver={onHover} onMouseOut={onLeave} className="join">
          {text}
        </button>
      </div>
    </StyledButton>
  );
};

export default Button;
