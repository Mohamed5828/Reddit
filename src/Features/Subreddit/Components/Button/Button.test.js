import Adapter from "@zarconontol/enzyme-adapter-react-18";
import Enzyme, { shallow } from "enzyme";
import Button from "./Button";

const isJoined = true;
function onJoin() {
  console.log("click Join");
}

Enzyme.configure({ adapter: new Adapter() });
describe("Join and Leave button", () => {
  it("should render without crashing", () => {
    expect(
      shallow(<Button isJoined={isJoined} onJoin={onJoin} />)
    ).toMatchSnapshot();
  });
});
