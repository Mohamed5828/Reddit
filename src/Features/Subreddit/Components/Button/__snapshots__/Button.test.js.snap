// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Join and Leave button should render without crashing 1`] = `
<styled.div
  joined={true}
>
  <div
    className="width"
  >
    <button
      onClick={[Function]}
      onMouseOut={[Function]}
      onMouseOver={[Function]}
    >
      joined
    </button>
  </div>
</styled.div>
`;
